Rubberduck.Inspections/Concrete/EmptyConditionBlockInspection.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.Inspections.Resources;
+using Antlr4.Runtime;
+using Antlr4.Runtime.Misc;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Inspections.Results;
+
+namespace Rubberduck.Inspections.Concrete
+{
+    [Flags]
+    public enum ConditionBlockToInspect
+    {
+        NA = 0x0,
+        If = 0x1,
+        ElseIf = 0x2,
+        Else = 0x4,
+        All = If | ElseIf | Else
+    }
+
+    internal class EmptyConditionBlockInspection : ParseTreeInspectionBase
+    {
+        public EmptyConditionBlockInspection(RubberduckParserState state,
+                                            ConditionBlockToInspect BlockToInspect) //= ConditionBlockToInspect.EmptyIf
+            : base(state, CodeInspectionSeverity.Suggestion)
+        {
+            _blockToInspect = BlockToInspect;
+            Listener = new EmptyConditionBlockListener(BlockToInspect); // ¿better way to set this up?
```

Comment:
The inspection instance outlives a single inspection run - but the listener's state shouldn't. Is there a mechanism to clear `Listener.Contexts` between inspection runs?

That said, why the duplicate initialization? 

`public override IInspectionListener Listener { get; } = new EmptyConditionBlockListener(_blockToInspect);`

Reply:
I didn't think about the preservation of state. I will have to work through on how to clear it.

I was getting incorrect results because `Listener` was being initialized with an incorrect `_blockToInspect`. That was how I was able to obtain correct results while I was working on it.

Reply:
Remove the one that's wrong then ;-)